Framework/Algorithms/src/SumOverlappingTubes.cpp

```diff
@@ -0,0 +1,366 @@
+      "A comma separated list of the first scattering angle, the scattering "
+      "angle step size and the final scattering angle. Optionally this can "
+      "also be a single number, which is the angle step size. In this case, "
+      "the boundary of binning will be determined by minimum and maximum "
+      "scattering angle present in the workspaces.");
+  declareProperty(make_unique<PropertyWithValue<std::string>>(
+                      "ComponentForHeightAxis", "tube_1", Direction::Input),
+                  "The name of the component to use for the height axis, that "
+                  "is the name of a PSD tube to be used. If specifying this "
+                  "then there is no need to give a value for the HeightBinning "
+                  "option.");
+  declareProperty(
+      make_unique<ArrayProperty<double>>(
+          "HeightAxis", boost::make_shared<RebinParamsValidator>(true, true)),
+      "A comma separated list of the first y value, the y value step size and "
+      "the final y value. This can also be a single number, which "
+      "is the y value step size. In this case, the boundary of binning will "
+      "be determined by minimum and maximum y values present in the "
+      "workspaces. For the 1DStraight case only this can also be two numbers, "
+      "to give the range desired.");
+  declareProperty(
+      make_unique<PropertyWithValue<bool>>("Normalise", true, Direction::Input),
+      "If true normalise to the number of entries added for a particular "
+      "scattering angle. If the maximum entries accross all the scattering "
+      "angles is N_MAX, and the number of entries for a scattering angle is N, "
+      "the normalisation is performed as N_MAX / N.");
+  auto toleranceValidator =
+      boost::make_shared<BoundedValidator<double>>(0.0, 0.0);
+  toleranceValidator->clearUpper();
+  declareProperty("ScatteringAngleTolerance", 0.0, toleranceValidator,
+                  "The relative tolerance for the scattering angles before the "
+                  "counts are split.");
+}
+
+std::map<std::string, std::string> SumOverlappingTubes::validateInputs() {
+  std::map<std::string, std::string> result;
+
+  const std::string componentForHeightAxis =
+      getProperty("ComponentForHeightAxis");
+  const std::string heightAxis = getProperty("HeightAxis");
+
+  if (componentForHeightAxis.empty() && heightAxis.empty()) {
+    std::string message =
+        "Either a component, such as a tube, must be specified "
+        "to get the height axis, or the binning given explicitly.";
+    result["ComponentForHeightAxis"] = message;
+    result["HeightBinning"] = message;
+  }
+
+  return result;
+}
+
+void SumOverlappingTubes::exec() {
+  getInputParameters();
+
+  HistogramData::Points x(m_numPoints, LinearGenerator(m_startScatteringAngle,
+                                                       m_stepScatteringAngle));
+
+  MatrixWorkspace_sptr outputWS = create<Workspace2D>(m_numHistograms, x);
```

Comment:
The output workspace doesn't seem to have an instrument attached. Is it intentional?

Reply:
Yes, that is intentional. Because the counts are put onto a regular 2D grid they are sometimes split, so keeping an instrument associated would not make sense anymore.